app/models/manageiq/providers/cloud_manager.rb

```diff
@@ -159,6 +159,8 @@ def self.display_name(number = 1)
 
     def destroy_mapped_tenants
       if source_tenant
+        # We just destroyed ourself, reload the source_tenant association
+        source_tenant.reload
         source_tenant.all_subtenants.destroy_all
         source_tenant.all_subprojects.destroy_all
```

Comment:
darn - feels like this should all be destroyed in tenant itself

Reply:
Yes, agreed